compiler/rustc_driver/src/session_diagnostics.rs

```diff
@@ -0,0 +1,17 @@
+use rustc_macros::SessionDiagnostic;
+
+#[derive(SessionDiagnostic)]
+#[diag(driver::rlink_unable_to_read)]
+pub(crate) struct RlinkUnableToRead {
+    pub error_message: String,
```

Comment:
See #100754 for an example of just including the `io::Error` directly in these types, which I think I prefer.